lib/puppet/util/tagging.rb

```diff
@@ -1,7 +1,7 @@
 require 'puppet/util/tag_set'
 
 module Puppet::Util::Tagging
-  ValidTagRegex = /\A[0-9A-Za-z_][0-9A-Za-z_:.-]*\Z/
+  ValidTagRegex = /\A[[:word:]][[[:word:]]:.-]*\Z/
```

Comment:
OK, but that's probably [PUP-4684](https://tickets.puppetlabs.com/browse/MODULES-4684), not 4864 ;-)